modules/programs/i3status.nix

```diff
@@ -0,0 +1,107 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.programs.i3status;
+
+  toStr = val:
+    if isString val then ''"${val}"''
+    else if isBool val then (if val then "true" else "false")
+    else toString val;
+
+  blockToConfig = block: ''${block.type} ${optionalString (block.name != null) ''"${block.name}"''} {
+    ${concatStringsSep "\n" (mapAttrsToList (k: v: "${k} = ${toStr v}") block.opts)}
+  }'';
+
+  configFile = pkgs.writeText "i3status.conf" ''
+    ${concatMapStringsSep "\n" (o: ''order += "${o}"'') cfg.order}
+    ${concatMapStringsSep "\n" blockToConfig (attrValues cfg.blocks)}
+  '';
+in {
+  options.programs.i3status = {
+    enable = mkEnableOption "i3status: Generates status bar to use with i3bar, dzen2 or xmobar.";
+
+    order = mkOption {
+      description = "Order of i3status blocks.";
+      type = types.listOf types.str;
+      default = [];
+      example = literalExample ''[
+        (mkOrder 500 "online_status")
+        (mkOrder 510 "disk")
+        (mkOrder 520 "load")
+        (mkOrder 530 "net_rate")
+        (mkOrder 540 "volume master")
+      ]'';
+    };
+
+    blocks = mkOption {
+      description = "Attribute set of defined i3status blocks.";
+      type = types.attrsOf (types.submodule ({name, config, ...}: {
```

Comment:
The `config` argument is not used, as far as I can tell. Perhaps safe to remove?